osu.Game/IO/StableStorage.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Linq;
+using osu.Framework.Platform;
+
+namespace osu.Game.IO
+{
+    /// <summary>
+    /// A storage pointing to an osu-stable installation.
+    /// Provides methods for handling installations with a custom Song folder location.
+    /// </summary>
+    public class StableStorage : DesktopStorage
+    {
+        private const string stable_songs_path = "Songs";
+
+        private readonly DesktopGameHost host;
+        private readonly string songsPath;
+
+        public StableStorage(string path, DesktopGameHost host)
+            : base(path, host)
+        {
+            this.host = host;
+            songsPath = locateSongsDirectory();
```

Comment:
not sure this should be initialised in the ctor, as opposed to on first use. there may be a case we initialise many `StableStorage` and this adds an overhead (it's doing directory listing and file parsing). A simple solution would be moving it to a `Lazy<string>`.